File: rest-service/src/main/java/mb/lib/query/model/UserBlastRow.java
package mb.lib.query.model;

import java.time.OffsetDateTime;

import mb.lib.model.HashID;
import mb.lib.model.JobStatus;
import org.veupathdb.lib.blast.BlastConfig;

public class UserBlastRow extends BlastRow
{
  private long    userID;
  private String  description;
  private long    maxDownloadSize;
  private boolean runDirectly;

  public UserBlastRow() {
  }

  public UserBlastRow(UserBlastRow old) {
    super(old);
    userID          = old.userID;
    description     = old.description;
    maxDownloadSize = old.maxDownloadSize;
    runDirectly     = old.runDirectly;
  }

  public UserBlastRow(BlastRow old) {
    super(old);
  }

  public long getUserID() {
    return userID;
  }

  public UserBlastRow setUserID(long userID) {
    this.userID = userID;
    return this;
  }

  public String getDescription() {
    return description;
  }

  public UserBlastRow setDescription(String description) {
    this.description = description;
    return this;
  }

  public long getMaxDownloadSize() {
    return maxDownloadSize;
  }

  public UserBlastRow setMaxDownloadSize(long maxDownloadSize) {
    this.maxDownloadSize = maxDownloadSize;
    return this;
  }

  public boolean isRunDirectly() {
    return runDirectly;
  }

  public UserBlastRow setRunDirectly(boolean runDirectly) {
    this.runDirectly = runDirectly;
    return this;
  }

  @Override
  public String toString() {
    return "UserBlastRow{jobID=" + getJobID() + ", userID=" + getUserID() + "}";
  }

  /// Chaining Overrides

  @Override
  public UserBlastRow setJobID(HashID jobID) {
    return (UserBlastRow) super.setJobID(jobID);
  }

  @Override
  public UserBlastRow setConfig(BlastConfig config) {
    return (UserBlastRow) super.setConfig(config);
  }

  @Override
  public UserBlastRow setQuery(String query) {
    return (UserBlastRow) super.setQuery(query);
  }

  @Override
  public UserBlastRow setQueueID(int queueID) {
    return (UserBlastRow) super.setQueueID(queueID);
  }

  @Override
  public UserBlastRow setProjectID(String projectID) {
    return (UserBlastRow) super.setProjectID(projectID);
  }

  @Override
  public UserBlastRow setStatus(JobStatus status) {
    return (UserBlastRow) super.setStatus(status);
  }

  @Override
  public UserBlastRow setCreatedOn(OffsetDateTime createdOn) {
    return (UserBlastRow) super.setCreatedOn(createdOn);
  }

  @Override
  public UserBlastRow setDeleteOn(OffsetDateTime deleteOn) {
    return (UserBlastRow) super.setDeleteOn(deleteOn);
  }
}

File: rest-service/src/main/kotlin/mb/lib/query/model/UserBlastRow.kt
package mb.lib.query.model

import mb.lib.model.JobStatus
import org.veupathdb.lib.hash_id.HashID
import java.time.OffsetDateTime

open class UserBlastRow(
  jobID:               HashID,
  config:              JobConfig?      = null,
  query:               String?         = null,
  queueID:             Int?            = null,
  projectID:           String?         = null,
  status:              JobStatus?      = null,
  createdOn:           OffsetDateTime? = null,
  deleteOn:            OffsetDateTime? = null,
  var userID:          Long,
  var description:     String?         = null,
  var maxDownloadSize: Long            = 0,
  var runDirectly:     Boolean         = false,
): BlastRow(jobID, config, query, queueID, projectID, status, createdOn, deleteOn) {

  constructor(
    old: BlastRow,
    userID: Long,
    description: String? = null,
    maxDownloadSize: Long = 0,
    runDirectly: Boolean = false,
  ) : this(
    jobID           = old.jobID,
    config          = old.config,
    query           = old.query,
    queueID         = old.queueID,
    projectID       = old.projectID,
    status          = old.status,
    createdOn       = old.createdOn,
    deleteOn        = old.deleteOn,
    userID          = userID,
    description     = description,
    maxDownloadSize = maxDownloadSize,
    runDirectly     = runDirectly
  )

  constructor(old: UserBlastRow): this(
    jobID           = old.jobID,
    config          = old.config,
    query           = old.query,
    queueID         = old.queueID,
    projectID       = old.projectID,
    status          = old.status,
    createdOn       = old.createdOn,
    deleteOn        = old.deleteOn,
    userID          = old.userID,
    description     = old.description,
    maxDownloadSize = old.maxDownloadSize,
    runDirectly     = old.runDirectly,
  )

  override fun toString(): String {
    return "UserBlastRow(jobID=${jobID}, userID=${userID})"
  }
}
